File: php/features.php
<?php
declare(strict_types=1);

// IsEven SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class IsEvenFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new IsEvenBaseFeature();
            case "test":
                return new IsEvenTestFeature();
            default:
                return new IsEvenBaseFeature();
        }
    }
}

File: py/iseven_sdk/features.py
# IsEven SDK feature factory

from iseven_sdk.feature.base_feature import IsEvenBaseFeature
from iseven_sdk.feature.test_feature import IsEvenTestFeature


def _make_feature(name):
    features = {
        "base": lambda: IsEvenBaseFeature(),
        "test": lambda: IsEvenTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
